gdal/apps/gdal_grid_lib.cpp

```diff
@@ -823,13 +832,27 @@ GDALDatasetH GDALGrid( const char *pszDest, GDALDatasetH hSrcDataset,
     if ( psOptions->pszSQL )
         nBands++;
 
-    // FIXME
-    int nXSize = psOptions->nXSize;
-    if ( nXSize == 0 )
-        nXSize = 256;
-    int nYSize = psOptions->nYSize;
-    if ( nYSize == 0 )
-        nYSize = 256;
+    int nXSize;
+    int nYSize;
+    if ( psOptions->dfXRes && psOptions->dfYRes )
+    {
+        nXSize = static_cast<int>(
+            (psOptions->dfXMax - psOptions->dfXMin + (psOptions->dfXRes/2.0)) /
+            psOptions->dfXRes);
+        nYSize = static_cast<int>(
+            (std::fabs(psOptions->dfYMax - psOptions->dfYMin) + (psOptions->dfYRes/2.0)) /
+            psOptions->dfYRes);
+    }
+    else
+    {
+        // FIXME
```

Comment:
The comment should explain what the `FIXME` is about

Reply:
This comment was already existing in the source. I don't know, what it is for. Sorry for that.

Reply:
Yes, indeed. The diff was a bit confusing.